### ISSUE TITLE
Missing OSGI package-exports from core-module

### ISSUE THREAD

**Issue by stoerr** [bbottema/simple-java-mail#288]
Hi!

Thank you for providing OSGI-headers in simple-java-mail to make it deployable as bundle! Unfortunately I get some trouble when trying to deploy it, together with it's required OSGI-bundles (core-module, jakarta.mail, jakarta.activation, emailaddress-rfc2822, jetbrains-runtime-annotations). I get some errors in the OSGI console (specifically, I'm using Felix in [Apache Sling](https://sling.apache.org/) ):

```
org.simplejavamail.api.internal.authenticatedsockssupport.socks5server -- Cannot be resolved
org.simplejavamail.api.internal.batchsupport -- Cannot be resolved
org.simplejavamail.api.internal.clisupport.model -- Cannot be resolved
org.simplejavamail.api.internal.outlooksupport.model -- Cannot be resolved
org.simplejavamail.api.internal.smimesupport.builder -- Cannot be resolved
org.simplejavamail.api.internal.smimesupport.model -- Cannot be resolved
org.simplejavamail.internal.util -- Cannot be resolved

```
It seems all these packages are contained in the core-module, which was successfully deployed and active. But it does not export those: it's exported packages are

```
org.simplejavamail,version=6.4.3
org.simplejavamail.api.email,version=6.4.3
org.simplejavamail.api.mailer,version=6.4.3
org.simplejavamail.api.mailer.config,version=6.4.3
org.simplejavamail.config,version=6.4.3

```
Could you please fix this? Thank you very much!

Best regards,

Hans-Peter Störr

**Comment by grueneerle** [comment]
Hi,

I stumbled upon the same problem, while trying to use simple-java-mail within an Apache Karaf Runtime Environment. Exports for internal packages are missing. I guess they're not exported autoamtically, because of the "*.internal.*" part in the package names. It's best practice to not export internal packages and the maven-bundle-plugin perhaps follow this practice if running with defaults.
The whole problem results from class loader isolation in OSGi. Every "bundle" (a deployable .jar with bundle information added to its MANIFEST.MF) resides in its own class loader and lookups to other bundles are controlled by package export/import declarations and dynamic dependency injection. Splitting a project in multiple modules whereas not providing these declaration makes it nearly impossible to use this project without restructuring the project.

Therefore I created a simple test project [simple-java-mail-osgi.zip](https://github.com/bbottema/simple-java-mail/files/5510271/simple-java-mail-osgi.zip) to re-bundle all needed simple-java-mail modules into one _osgi_ bundle with complete package export/import declarations. There are just some extra instructions for the _maven-bundle-plugin_ needed to achieve this.
While inspecting the resulting artifact, I recognized an auto generated import declaration `org.jacoco.agent.rt.internal_c13123e` alongside some other requirements and thought they came from some "auto import" magic and prohibited their creation by adding it to the _maven-bundle-plugin_ instructions with a leading `!` in front of it.

The bundle deployed and resolved straightforward and I start trying to use it. But when trying to compose an email via
` EmailPopulatingBuilder emailBuilder = EmailBuilder.startingBlank();`
the runtime crashes resulting in a
`java.lang.NoClassDefFoundError: org/jacoco/agent/rt/internal_c13123e/Offline` (Stacktrace attached)

Why is simple-java-mail relying on the jacoco offline agent during runtime? Is this a result of the jacoco `instrument` usage?

I hope someone could check my attached solution and acknowledges the results. Thanks!

Kind regards,
Rene Greuel

Stacktrace: [stacktrace.txt](https://github.com/bbottema/simple-java-mail/files/5509981/stacktrace.txt)
